### PR TITLE
Improve PHP 8.2 compatibility and testing

### LINKED ISSUE #164: PHPUnit warning: assertRegExp is deprecated in favour of assertMatchesRegularExpression

### PR DESCRIPTION
**Pull request by AndyTWF**
- Adds PHP 8.2 compatibility without platform-reqs concessions
- Updates a flakey test to allow for differences between CI and Server time
- Removes use of deprecated PHPUnit method (and introduces a wrapper that we can remove once PHP 7.2 is dropped from CI)

Fixes #164

### CODE REVIEW

**Comment by sacOO7** [comment]
Can you update file name for `check.yml` to `build-test.yml` ?

**Comment by sacOO7** [comment]
Also, check if there is a need to update `composer.json`.

**Comment by AndyTWF** [comment]
I believe that `check.yml` is consistent with the filename conventions in https://github.com/ably/engineering/blob/main/sdk/github.md#filenames

**Comment by AndyTWF** [comment]
`composer.json` appears to be fine - but will double check

File: tests/AssertsRegularExpressions.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    public static function assertMatchesRegularExpression(string $pattern, string $string, string $message = ''): void
+    {
+        if (version_compare(phpversion(), '7.3.0', '<')) {
+            self::assertRegExp($pattern, $string, $message);
+            return;
+        }
+
+        parent::assertMatchesRegularExpression($pattern, $string, $message);
+    }
+}
```
Comment (owenpearson) [inline tests/AssertsRegularExpressions.php:25]: probably should add a new line at the end here

**Review by owenpearson — APPROVED** [review]
LGTM, thanks

File: tests/AssertsRegularExpressions.php (hunk 1 above)
Comment (AndyTWF) [reply tests/AssertsRegularExpressions.php]: Done!

**Pull request merged**